monkey/monkey_island/cc/ui/src/components/report-components/zerotrust/VennDiagram/index.js

```diff
@@ -0,0 +1,369 @@
+    
+  constructor(props) { super(props); }
+  render() {
+      
+    const {options, pillarsGrades} = this.props;
+    let childrenWidth = this.props.containerWidth,  childrenHeight = this.props.containerHeight;
+      
+    if(childrenHeight === 0 ||  childrenHeight === NaN){  childrenHeight = childrenWidth; } 
+    else{ childrenWidth = Math.min(childrenWidth, childrenHeight) }
+
+    return ( 
+        
+            <div ref={this.divElement}> 
+                <VennDiagram width={Math.max(childrenWidth, VENN_MIN_WIDTH)} height={Math.max(childrenHeight, VENN_MIN_WIDTH)} pillarsGrades={pillarsGrades} />
+            </div> )
+
+  }
+            
+}
+
+ResponsiveVennDiagram.propTypes = {
+        
+      pillarsGrades: PropTypes.array
+        
+}
+
+export default Dimensions()(ResponsiveVennDiagram);
+
+class VennDiagram extends React.Component{
+    
+    constructor(props_){ 
+  
+        super(props_); 
+        
+        this.state = { tooltip: { top: 0, left: 0, display: 'none', html: '' } }
+        
+        this.colors = ['#777777', '#D9534F', '#F0AD4E', '#5CB85C'];
+        this.prefix = 'vennDiagram';
+        this.suffices = ['', '|tests are|conclusive', '|tests were|inconclusive', '|tests|performed'];
+        this.fontStyles = [{size: Math.max(9, this.props.width / 32), color: 'white'}, {size: Math.max(6, this.props.width / 52), color: 'black'}];
+        this.offset = this.props.width / 16;
+        
+        this.thirdWidth = this.props.width / 3; 
+        this.sixthWidth = this.props.width / 6;
+        this.width2By7 = 2 * this.props.width / 7
+        this.width1By11 = this.props.width / 11;
+        this.width1By28 = this.props.width / 28;
+        
+        this.toggle = false;
+        
+        this.layout = {
+            
+            Data: { cx: 0, cy: 0, r: this.thirdWidth - this.offset * 2, offset: {x: 0, y: 0} },
+            People: { cx: -this.width2By7, cy: 0, r: this.sixthWidth, offset: {x: this.width1By11, y: 0} },
+            Networks: { cx: this.width2By7, cy: 0, r: this.sixthWidth, offset: {x: -this.width1By11, y: 0} },
+            Devices: { cx: 0, cy: this.width2By7, r: this.sixthWidth, offset: {x: 0, y: -this.width1By11} },
+            Workloads : { cx: 0, cy: -this.width2By7, r: this.sixthWidth, offset: {x: 0, y: this.width1By11} },
+            VisibilityAndAnalytics : { inner: this.thirdWidth - this.width1By28, outer: this.thirdWidth },
+            AutomationAndOrchestration: { inner: this.thirdWidth - this.width1By28 * 2, outer: this.thirdWidth - this.width1By28 }
+
+        };
+    
+    }
+    
+    componentDidMount() {
+
+        this.parseData();
+        
+    }
+
+     _onMouseMove(e) {
+         
+        if(!this.toggle){
+            
+            let hidden = 'none';
+            let html = '';
+            let bcolor = '#DEDEDE';
+
+            e.stopPropagation();
+
+            document.querySelectorAll('circle, path').forEach((d_, i_) => { d_.setAttribute('opacity', 0.8)});
```

Comment:
Add documentation to this and the previous lines

Reply:
You're right, I don't need this method for click event in mousemove function. More likely I have misplaced it while coding onClick function and experimenting with callbacks.